### PR TITLE
Upgrade to Hibernate ORM 6, Search 6 and Jakarta Persistence 3.1

### PR DESCRIPTION
**Pull request by scottmarlow**
Signed-off-by: Scott Marlow <smarlow@redhat.com>
https://issues.redhat.com/browse/EAP7-1815
https://issues.redhat.com/browse/EAP7-1816

### CODE REVIEW

File: jpa/WFLY-15440_Persistence3.1_Hibernate6.0.adoc [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+
+Included JPA 3.1, Hibernate ORM 6.0+, Hibernate Search 6.1+.
+
+== Test Plan
+
+The Jakarta EE 10 TCKs must be passed with the upgraded Hibernate ORM version.
+
+The Hibernate ORM tests in the WildFly testsuite must be passed.
+
+The Hibernate ORM tests in the Hibernate ORM project must be passed against the following database servers for EAP 8:
```
Comment (simkam) [inline jpa/WFLY-15440_Persistence3.1_Hibernate6.0.adoc:52]: Let's wait whether Brad has some feedback but currently [preferred versions](https://issues.redhat.com/browse/EAP7-1820?focusedCommentId=20571783&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-20571783) are LTS versions of Oracle and MariaDB.

Oracle (RAC) version 19c
MariaDB (+ Galera) 10.6

File: jpa/WFLY-15440_Persistence3.1_Hibernate6.0.adoc [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+:idprefix:
+:idseparator:       -
+
+== Overview
+
+Support Persistence 3.1 by including Hibernate ORM 6.0.x https://issues.redhat.com/browse/WFLY-15440/[WFLY-15440] + https://issues.redhat.com/browse/EAP7-1815[EAP7-1815].  Also update the WildFly Persistence Container to support Jakarta Persistence 3.1.
+
+The list of databases to test with are in comments on https://issues.redhat.com/browse/EAP7-1820.
+
+Consider including Hibernate https://github.com/hibernate/hibernate-orm/releases/tag/6.1.0[ORM 6.1.0] instead of Hibernate ORM 6.0.x.
```
Comment (bstansberry) [inline jpa/WFLY-15440_Persistence3.1_Hibernate6.0.adoc:15]: Is this a requirement now, not just a consideration?

AIUI, moving to 6.1.x  is required to:

1) Resolve security manager failures: https://issues.redhat.com/browse/WFLY-16552

2) Support Jandex 3, which is both desirable in general for WF, but is also a requirement for https://issues.redhat.com/browse/WFLY-16959, the move of Smallrye Open API to a .Final version, which is a blocker.

Comment (bstansberry) [reply jpa/WFLY-15440_Persistence3.1_Hibernate6.0.adoc:15]: Answering myself; it is a requirement now.

**Comment by vsalbaba** [comment]
Note the PR https://github.com/wildfly/wildfly-proposals/pull/503 which would bring Hibernate ORM to 6.2

**Comment by scottmarlow** [comment]
Closing this pr as out of date and replaced by https://github.com/wildfly/wildfly-proposals/pull/503

**Pull request closed without merging**